rules-java/api/pom.xml

```diff
@@ -35,6 +35,23 @@
             <version>5.0.4</version>
         </dependency>
 
+        <!-- Local Dependencies -->
+        <dependency>
+            <groupId>org.jboss.windup.maven</groupId>
+            <artifactId>nexus-indexer-data</artifactId>
+            <version>6-SNAPSHOT</version>
+            <scope>compile</scope>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.lucene</groupId>
+            <artifactId>lucene-core</artifactId>
+            <version>4.8.1</version>
```

Comment:
I agree that we should better keep things isolated to addons.
If it happens that same dependency appears in more than, say, 5 addons, we could create another addon just for that dependency.
BOM may define those which go throughout the project, like commons-lang.
See also WINDUP-1007.